### PR TITLE
[2.2.2-patch]Support helm install with app info

### PR DESCRIPTION
**Pull request by orangedeng**
Inject cluster name, project name and default registy parameters as
extra answers values when helm installing.
We are no longer storing them in app answers like the old default
registry parameter.

### CODE REVIEW

File: pkg/controllers/user/helm/common/common.go [hunk 1]
```diff
@@ -162,10 +137,16 @@ func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
...
 	// `--set` values will overridden the user-supplied values.yaml file
-	if app.Spec.Answers != nil {
+	if app.Spec.Answers != nil || extraArgs != nil {
 		answers := app.Spec.Answers
 		var values = []string{}
 		for k, v := range answers {
+			if _, ok := extraArgs[k]; ok {
+				continue
+			}
+			values = append(values, fmt.Sprintf("%s=%s", k, v))
```
Comment (alena1108) [inline pkg/controllers/user/helm/common/common.go]: do we need to sort the values?

File: pkg/controllers/user/helm/common/extra_args_injection.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+func injectDefaultRegistry(obj *v3.App) map[string]string {
+	values, err := url.Parse(obj.Spec.ExternalID)
+	if err != nil {
+		logrus.Errorf("check catalog type failed: %s", err.Error())
+	}
+
+	catalogWithNamespace := values.Query().Get("catalog")
+	split := strings.SplitN(catalogWithNamespace, "/", 2)
```
Comment (alena1108) [inline pkg/controllers/user/helm/common/extra_args_injection.go]: shall we check the len of split? Just to be safe

File: pkg/controllers/user/helm/common/extra_args_injection.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	catalog := split[len(split)-1]
+
+	reg := settings.SystemDefaultRegistry.Get()
+	if catalog != systemCatalogName || reg == "" {
```
Comment (alena1108) [inline pkg/controllers/user/helm/common/extra_args_injection.go]: `catalog != systemCatalogName` can be done before we fetch the registry from etcd. This way you save on one call if the first condition is true

File: pkg/controllers/user/helm/common/common.go (hunk 1 above)
Comment (orangedeng) [reply pkg/controllers/user/helm/common/common.go]: I don't think we need. Helm takes all the `--set` values as input. The sequence fo the `--set` values won't change the result of `helm install`.

**Comment by orangedeng** [comment]
Closing this PR because we have same commit in master and release/v2.2 branches. Have small fixes of above comments in separate PRs.

**Pull request closed without merging**